src/go/k8s/internal/controller/redpanda/managed_decommission_controller.go

```diff
@@ -0,0 +1,580 @@
+		return &resources.RequeueAfterError{
+			RequeueAfter: wait.Jitter(decommissionWaitInterval, decommissionWaitJitterFactor),
+			Msg:          fmt.Sprintf("broker %d decommission status not finished", decommissionNodeID),
+		}
+	}
+
+	log.Info("Node decommissioned")
+	er.AnnotatedEventf(rp,
+		map[string]string{v1alpha1.GroupVersion.Group + revisionPath: rp.ResourceVersion},
+		corev1.EventTypeNormal, v1alpha1.EventTypeTrace, fmt.Sprintf("Node decommissioned: %d", decommissionNodeID))
+
+	err = podEvict(ctx, log, c, rp)
+	if err != nil {
+		return err
+	}
+
+	rp.Status.ManagedDecommissioningNode = nil
+
+	return updateRedpanda(ctx, rp, c)
+}
+
+var ErrZeroReplicas = errors.New("redpanda replicas is zero")
+
+func getRedpandaReplicas(ctx context.Context, c k8sclient.Client, rp *v1alpha1.Redpanda) (int, error) {
+	requestedReplicas := ptr.Deref(rp.Spec.ClusterSpec.Statefulset.Replicas, 0)
+	if requestedReplicas == 0 {
+		resourcesName := rp.Name
+		if rp.Spec.ClusterSpec.FullNameOverride != "" {
+			resourcesName = rp.Spec.ClusterSpec.FullNameOverride
+		}
+
+		var sts appsv1.StatefulSet
+		err := c.Get(ctx, types.NamespacedName{
+			Namespace: rp.Namespace,
+			Name:      resourcesName,
+		}, &sts)
+		if err != nil {
+			return 0, fmt.Errorf("getting statefulset : %w", err)
+		}
+
+		requestedReplicas = int(ptr.Deref(sts.Spec.Replicas, 0))
+	}
+
+	if requestedReplicas == 0 {
+		return 0, ErrZeroReplicas
+	}
+	return requestedReplicas, nil
+}
+
+func podEvict(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha1.Redpanda) error {
+	if rp.Status.ManagedDecommissioningNode == nil {
+		return nil
+	}
+
+	decommissionNodeID := int(*rp.Status.ManagedDecommissioningNode)
+	log := l.WithName("podEvict").WithValues("decommission-node-id", decommissionNodeID)
+
+	pod, err := getPodFromRedpandaNodeID(ctx, l, c, rp, decommissionNodeID)
+	if err != nil {
+		return fmt.Errorf("getting pod from node-id (%d): %w", decommissionNodeID, err)
+	}
+
+	log.Info("delete Pod PVC", "pod-name", pod.Name)
+	if err = utils.DeletePodPVCs(ctx, c, pod, log); err != nil {
+		return fmt.Errorf(`unable to remove VPCs for pod "%s/%s: %w"`, pod.GetNamespace(), pod.GetName(), err)
+	}
+	return c.Delete(ctx, pod)
+}
+
+func getPodFromRedpandaNodeID(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha1.Redpanda, nodeID int) (*corev1.Pod, error) {
+	log := l.WithName("getPodFromRedpandaNodeID")
+
+	pl, err := getPodList(ctx, c, rp.Namespace, labels.Set{
+		"app.kubernetes.io/component": componentLabelValue,
+		"app.kubernetes.io/instance":  rp.Name,
+	}.AsSelector())
+	if err != nil {
+		return nil, fmt.Errorf("check all pods have condition: %w", err)
+	}
+
+	valuesMap, err := getHelmValues(l, rp.GetHelmReleaseName(), rp.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("get helm values: %w", err)
+	}
+
+	for i := range pl.Items {
+		pod := pl.Items[i]
+
+		log.WithValues("pod-name", pod.Name)
+
+		if !utils.IsPodReady(&pod) {
```

Comment:
I think i understand the not ready case logging it, but why return from here? what about other pods?

Reply:
> why return from here?

I'm returning with special error named `RequeueAfterError`.

```go
                        &resources.RequeueAfterError{
				RequeueAfter: wait.Jitter(decommissionWaitInterval, decommissionWaitJitterFactor),
				Msg:          "pod is not ready",
			}
```

https://github.com/redpanda-data/redpanda-operator/pull/49/files#diff-3b50bc37a129716ec058fcced9e244f5a17a55f4d26f5d1919ebcde3773f9d28R309-R312

That is handled 
```
	if err = decommissionStatus(ctx, log, c, er, rp); err != nil {
		var requeueErr *resources.RequeueAfterError
		if errors.As(err, &requeueErr) {
			log.Info(requeueErr.Error())
			return ctrl.Result{RequeueAfter: requeueErr.RequeueAfter}, nil
		}
		return ctrl.Result{}, fmt.Errorf("managed decommission status: %w", err)
	}
```

https://github.com/redpanda-data/redpanda-operator/pull/49/files#diff-3b50bc37a129716ec058fcced9e244f5a17a55f4d26f5d1919ebcde3773f9d28R121

>  what about other pods?

The reconciliation will be retried after `wait.Jitter(decommissionWaitInterval, decommissionWaitJitterFactor)`.